### PR TITLE
WINC-602: [wmco] Support Windows Server 2022

### PR DESCRIPTION
**Pull request by saifshaikh48**
This PR removes SAC Windows Server 2004 (EOL as of 2021-12-14) from our CI and docs,
and replaces it with LTSC Windows Server 2022 
(support announced here kubernetes/kubernetes#104438, EOL is 2026-10-13).
It also improves documentation around building the vSphere golden image using Packer.

Windows Server 2022 is tested in the vSphere environments. 
The golden image is built with a private networking OS fix supplied my Microsoft, 
so official support is on hold until they release the patch publicly.
Support on AWS and Azure is not included in this work.

### CODE REVIEW

**Comment by saifshaikh48** [comment]
Needs https://github.com/openshift/windows-machine-config-bootstrapper/pull/316 to merge and subsequent WMCB submodule update.

File: docs/vsphere_ci/build.json [hunk 1]
```diff
@@ -50,6 +50,7 @@
         "scripts/install-docker.ps1",
         "scripts/install-firewall-rules.ps1"
       ],
+      "vm_version": 15,
```
Comment (jrvaldes) [inline docs/vsphere_ci/build.json:53]: This should go in a separate PR. See [WINC-726](https://issues.redhat.com/browse/WINC-726)

File: docs/vsphere_ci/README.md [hunk 2]
```diff
@@ -14,7 +14,7 @@ to be executed from the bastion host or any instance that has access to the vSph
 
 ## Installing Packer
 
-In case of our vSphere environment, we need to install Packer 1.6.6 on the bastion host. The installation steps are:
+In case of our vSphere environment, we need to install Packer 1.6.6+ on the bastion host. The installation steps are:
```
Comment (jrvaldes) [inline docs/vsphere_ci/README.md:17]: I don't think this is needed here. What Packer version did you test? 

Relaxing the Packer version with a lower version constrain may have other implications. For instance, we should point the installation details to a generic page in Packer documentation, instead of [suggesting](https://github.com/openshift/windows-machine-config-operator/blob/912e92a5f09dcd7925488e8958b687dee416454e/docs/vsphere_ci/README.md?plain=1#L20) a fixed version.

Comment (saifshaikh48) [reply docs/vsphere_ci/README.md:17]: > instead of suggesting a fixed version

I added the `+` to this line for this reason. We tested with 1.6.6 in the past (and I used it for this golden image too) but there are newer released versions of Packer. I can remove the version suggestion entirely if that'd be better?

File: docs/vsphere_ci/build.json (hunk 1 above)
Comment (saifshaikh48) [reply docs/vsphere_ci/build.json]: Opened https://github.com/openshift/windows-machine-config-operator/pull/895

File: docs/vsphere_ci/README.md [hunk 3]
```diff
@@ -99,6 +100,16 @@ To enable detailed logging:
...
 
+### What to do during the Packer build 
+
+During the golden image creation, it is highly recommended to establish access to the virtual machine by launching a
+Web Console through the vSphere Client. This can be done after the Packer build has powered on the VM (while it is
+<i>Waiting for IP...</i>). 
+
+If the build halts and prompts for a product key during the Windows OS setup, manual intervention will be required.
+When accessing the golden image via Web Console, send a `Ctrl+Alt+Del` then tab over to `I don't have a product key`,
+and hit enter on the keyboard. This should start the OS setup as intended.
```
Comment (jrvaldes) [inline docs/vsphere_ci/README.md:111]: Is this required even after [providing](https://github.com/openshift/windows-machine-config-operator/pull/892/commits/848749c1def7ab9435837f9abd25134dd1ebee00#diff-f95d5d0ee98dedf392fb54c0fb93f7e3df3c9699b01d08ca8fd00d30ccba8244R66) the `ProductKey`?

Comment (saifshaikh48) [reply docs/vsphere_ci/README.md:111]: It was for both Sebastian and I.

File: hack/machineset.sh [hunk 4]
```diff
@@ -185,7 +185,7 @@ get_vsphere_ms() {
 
   # set golden image template name
   # TODO: read from parameter
-  template="windows-golden-images/windows-server-2004-template"
+  template="windows-golden-images/windows-server-2022-template"
```
Comment (jrvaldes) [inline hack/machineset.sh:188]: We need to update it [here](https://github.com/openshift/release/pull/24911/files#diff-c720c0075993159417e4f67b269d3d26696299d9db2409d49d0928ce660ee6d3R14) too, when https://github.com/openshift/release/pull/24911 merges.

Comment (saifshaikh48) [reply hack/machineset.sh:188]: True, noted :writing_hand:

File: docs/vsphere-golden-image.md [hunk 5]
```diff
@@ -4,13 +4,16 @@ This guide describes the thought process of creating a Windows virtual machine b
 
 ## 1. Select a compatible Windows Server version
 
-Currently, the Windows Machine Config Operator (WMCO) stable version only supports Windows Server Semi-Annual
-Channel (SAC): Windows Server 2004, that includes the patch [KB4565351](https://support.microsoft.com/en-us/help/4565351/windows-10-update-kb4565351), 
-required by the operating system to allow using the [hybrid OVN Kubernetes networking with a 
-custom VXLAN port](setup-hybrid-OVNKubernetes-cluster.md#vSphere) feature.
+Currently, the Windows Machine Config Operator (WMCO) stable version supports Windows Server Long-Term Servicing Channel
+(LTSC): Windows Server 2022 and Windows Server Semi-Annual Channel (SAC): Windows Server 20H2.
```
Comment (sebsoto) [inline docs/vsphere-golden-image.md:8]: nit: the presentation of these versions could make it easy to skim through
consider using bullet points to have the versions take up more space and draw more attention
i.e:



Currently, the Windows Machine Config Operator (WMCO) stable version supports:
* Windows Server 2022
* Windows Server 20H2

Comment (sebsoto) [reply docs/vsphere-golden-image.md:8]: Additionally the `Long-Term Servicing Channel` /`SAC` information isn't really needed IMO, just adds more noise

File: docs/vsphere-golden-image.md [hunk 6]
```diff
@@ -4,13 +4,16 @@ This guide describes the thought process of creating a Windows virtual machine b
...
+
+*Please note that Windows Server Semi-Annual Channel (SAC): Windows Server 2004 is no longer supported. 
+Also, Windows Server Long-Term Servicing Channel (LTSC): Windows Server 1809 cannot be used since patch 
+[KB4565351](https://support.microsoft.com/en-us/help/4565351/windows-10-update-kb4565351) is not available, which is
+required by the operating system to allow using the 
+[hybrid OVN Kubernetes networking with a custom VXLAN port](setup-hybrid-OVNKubernetes-cluster.md#vSphere) feature.*
```
Comment (sebsoto) [inline docs/vsphere-golden-image.md:14]: As above LTSC is not important here, but some important info that its missing is that this is actually talking about Windows Server 2019.

This can be condensed to:
Please note that Windows Server 2004 is no longer supported. 
Also, Windows Server 2019 cannot be used since patch  [KB4565351](https://support.microsoft.com/en-us/help/4565351/windows-10-update-kb4565351) is not available, which is required by the operating system to allow using the [hybrid OVN Kubernetes networking with a custom VXLAN port](setup-hybrid-OVNKubernetes-cluster.md#vSphere) feature

File: docs/vsphere_ci/README.md (hunk 3 above)
Comment (sebsoto) [reply docs/vsphere_ci/README.md:111]: Maybe a user will figure out whats going on here and open an issue helping us out :)

**Review by sebsoto — CHANGES_REQUESTED** [review]

File: docs/vsphere-golden-image.md (hunk 6 above)
Comment (saifshaikh48) [reply docs/vsphere-golden-image.md:14]: I thought the note was intentional with Windows Server 2019, version 1809 (which is still in service) since Windows Server 2019, 1909 version does have the linked patch (but is EOL).

I can make it a bit more explicit by saying `Windows Server 2019, version 1809 (LTSC)`.

**Comment by sebsoto** [comment]
/hold
in case nit changed

File: docs/vsphere-golden-image.md [hunk 7]
```diff
@@ -4,13 +4,17 @@ This guide describes the thought process of creating a Windows virtual machine b
...
-Channel (SAC): Windows Server 2004, that includes the patch [KB4565351](https://support.microsoft.com/en-us/help/4565351/windows-10-update-kb4565351), 
-required by the operating system to allow using the [hybrid OVN Kubernetes networking with a 
-custom VXLAN port](setup-hybrid-OVNKubernetes-cluster.md#vSphere) feature.
-
-*Please note that Windows Server Long-Term Servicing Channel (LTSC): Windows Server 1809 cannot be used, since 
-the patch is not available.*
+Currently, the Windows Machine Config Operator (WMCO) stable version supports:
+<ul>
+    <li>Windows Server 2022</li>
+    <li>Windows Server 20H2</li>
```
Comment (jrvaldes) [inline docs/vsphere-golden-image.md:10]: What about just adding a pointer to [Supported Windows Server versions](https://github.com/openshift/windows-machine-config-operator/blob/master/docs/wmco-prerequisites.md#supported-windows-server-versions)? It Makes things simpler and avoid content duplication.

File: docs/vsphere_ci/README.md (hunk 3 above)
Comment (jrvaldes) [reply docs/vsphere_ci/README.md:111]: I follow, manual intervention is required. Why adding the `ProductKey` section in [autounattend.xml](https://github.com/openshift/windows-machine-config-operator/pull/892/commits/9826af4103d03e40a0d2d8466854c3e341a0d700#diff-f95d5d0ee98dedf392fb54c0fb93f7e3df3c9699b01d08ca8fd00d30ccba8244).

File: docs/vsphere-golden-image.md [hunk 8]
```diff
@@ -4,13 +4,17 @@ This guide describes the thought process of creating a Windows virtual machine b
...
+</ul>
```
[the remark this answers is not in the document]
Comment (saifshaikh48) [reply docs/vsphere-golden-image.md:11]: thanks still learning markdown :slightly_smiling_face:

File: docs/vsphere-golden-image.md (hunk 7 above)
Comment (saifshaikh48) [reply docs/vsphere-golden-image.md:10]: I think its worth keeping here explicitly given [the extra explanation](https://github.com/openshift/windows-machine-config-operator/pull/892/files#diff-eff7340e15df41ff3a96507abf744eb5fea3f670fbb381edc624b52cda6b1b79R13-R17) we give. The golden image steps are somewhat complex as is so I'd like to avoid sending them off this page.

File: hack/machineset.sh (hunk 4 above)
Comment (selansen) [reply hack/machineset.sh:188]: Thanks for pointing out @jrvaldes . do we have jira ticket to track this?

Comment (saifshaikh48) [reply hack/machineset.sh:188]: @selansen added to acceptance criteria of https://issues.redhat.com/browse/WINC-602

File: docs/vsphere_ci/README.md (hunk 3 above)
Comment (jrvaldes) [reply docs/vsphere_ci/README.md:111]: I suggest removing it, in case the ProductKey section is not required in [autounattend.xml](https://github.com/openshift/windows-machine-config-operator/pull/892/commits/9826af4103d03e40a0d2d8466854c3e341a0d700#diff-f95d5d0ee98dedf392fb54c0fb93f7e3df3c9699b01d08ca8fd00d30ccba8244).

File: docs/wmco-prerequisites.md [hunk 9]
```diff
@@ -40,7 +40,7 @@ Note: OpenShiftSDN networking is the default network for OpenShift clusters. It
 | Hybrid OVNKubernetes      | Supported Windows Server version                                      |
 | -----------               | -----------                                                           |
 | Default VXLAN port        | Windows Server Long-Term Servicing Channel (LTSC): Windows Server 2019|
-| Custom VXLAN port         | Windows Server Semi-Annual Channel (SAC): Windows Server 20H2|
+| Custom VXLAN port         | <ul><li>Windows Server Long-Term Servicing Channel (LTSC): Windows Server 2022</li><li>Windows Server Semi-Annual Channel (SAC): Windows Server 20H2</li></ul>|
```
Comment (aravindhp) [inline docs/wmco-prerequisites.md:43]: Please stick to the existing formating
```suggestion
| Custom VXLAN port         | Windows Server Long-Term Servicing Channel (LTSC): Windows Server 2022|
|                                           | Windows Server Semi-Annual Channel (SAC): Windows Server 20H2|

File: docs/wmco-prerequisites.md [hunk 10]
```diff
@@ -40,7 +40,7 @@ Note: OpenShiftSDN networking is the default network for OpenShift clusters. It
 | Hybrid OVNKubernetes      | Supported Windows Server version                                      |
 | -----------               | -----------                                                           |
 | Default VXLAN port        | Windows Server Long-Term Servicing Channel (LTSC): Windows Server 2019|
```
Comment (aravindhp) [inline docs/wmco-prerequisites.md:42]: Please add 2022 support here also.

**Review by aravindhp — CHANGES_REQUESTED** [review]
@saifshaikh48 we are adding 2022 support across cloud providers and not just vSphere, so please reflect that. If that results in the question of testing 2022 in each provider, please note we have not been doing that for every Windows variant in the past either.

File: docs/wmco-prerequisites.md (hunk 9 above)
Comment (aravindhp) [reply docs/wmco-prerequisites.md:43]: Or something to this effect without introducing any HTML tags.

Comment (saifshaikh48) [reply docs/wmco-prerequisites.md]: Tried to make things look decent without any embedded tags. LMK if it's clear.

Comment (aravindhp) [reply docs/wmco-prerequisites.md]: Markdown all the way 👍🏽

**Comment by aravindhp** [comment]
@saifshaikh48 please remove `in vSphere` in  [[docs] Support Windows Server 2022 in vSphere](https://github.com/openshift/windows-machine-config-operator/pull/892/commits/0017c9684d9d55b1434a908d82fa0683d967efa7) as we are adding it for all supported platforms.

File: docs/vsphere_ci/README.md (hunk 3 above)
Comment (saifshaikh48) [reply docs/vsphere_ci/README.md:111]: How do we know removing it will not break golden image creation though?

Comment (jrvaldes) [reply docs/vsphere_ci/README.md:111]: Testing the golden image creation without the ProductKey section in [autounattend.xml](https://github.com/openshift/windows-machine-config-operator/pull/892/commits/9826af4103d03e40a0d2d8466854c3e341a0d700#diff-f95d5d0ee98dedf392fb54c0fb93f7e3df3c9699b01d08ca8fd00d30ccba8244).

File: README.md [hunk 11]
```diff
@@ -117,8 +117,8 @@ The following template variables need to be replaced as follows with values from
 * *\<vCenter Server FQDN/IP\>*: IP address or FQDN of the vCenter server
 
 *IMPORTANT*:
-- The template used in the MachineSet must be a Windows Server 1909
-  image as described in [vSphere prerequisites](docs/vsphere-prerequisites.md).
+- The template used in the MachineSet must be an image with a supported Windows Server
+  version, as described in [vSphere prerequisites](docs/vsphere-prerequisites.md).
```
Comment (sebsoto) [inline README.md:121]: This wording seems off. How about something like this?

`The VM template provided in the MachineSet must use a supported Windows Server version...`

Basing this off [this doc](https://docs.vmware.com/en/VMware-vSphere/6.0/com.vmware.vsphere.hostclient.doc/GUID-F40130B0-0194-4A41-91FA-1A967721924B.html), in that a template is a master copy of a VM, so the `image` language used seems a bit redundant.

**Review by sebsoto — CHANGES_REQUESTED** [review]

File: docs/vsphere_ci/README.md (hunk 3 above)
Comment (saifshaikh48) [reply docs/vsphere_ci/README.md:111]: When creating template without ProductKey in autounattend.xml, golden image VM fails to complete Microsoft Server Operating System Setup with message: `Windows cannot read the <ProductKey> setting from the unattend answer file`

File: hack/machineset.sh (hunk 4 above)
Comment (saifshaikh48) [reply hack/machineset.sh:188]: PR opened here: https://github.com/openshift/release/pull/26435

**Comment by saifshaikh48** [comment]
/test vsphere-e2e-operator

I am not seeing the same SSH connection failures when testing BYOH node config locally with cluster version of 4.11.0-0.nightly-2022-02-16-131355

**Comment by jrvaldes** [comment]
/hold
while azure CI is resolved

File: docs/wmco-prerequisites.md [hunk 12]
```diff
@@ -20,10 +20,8 @@ these errors, only use the appropriate version according to the cloud provider i
 
 | Cloud Provider      | Supported Windows Server version                                                     |
 | -----------         | -----------                                                                          |
-| AWS                 | Windows Server Long-Term Servicing Channel (LTSC): Windows Server 2022, version 21H2 |
-|                     | Windows Server Long-Term Servicing Channel (LTSC): Windows Server 2019               |
-| Azure               | Windows Server Long-Term Servicing Channel (LTSC): Windows Server 2022, version 21H2 |
-|                     | Windows Server Long-Term Servicing Channel (LTSC): Windows Server 2019               |
+| AWS                 | Windows Server Long-Term Servicing Channel (LTSC): Windows Server 2019, version 1809 |
+| Azure               | Windows Server Long-Term Servicing Channel (LTSC): Windows Server 2019, version 1809 |
```
Comment (saifshaikh48) [inline docs/wmco-prerequisites.md:24]: I'll squash these `.md` changes into the `[docs]` commit after a round of review.

**Comment by saifshaikh48** [comment]
/hold

for https://github.com/openshift/release/pull/26435, given we want to test Windows Server 2022 on platoform-none as well

File: pkg/windows/windows.go [hunk 13]
```diff
@@ -71,8 +71,8 @@ const (
 	windowsExporterServiceArgs = "--collectors.enabled " +
 		"cpu,cs,logical_disk,net,os,service,system,textfile,container,memory,cpu_info"
 	// remotePowerShellCmdPrefix holds the PowerShell prefix that needs to be prefixed  for every remote PowerShell
-	// command executed on the remote Windows VM
-	remotePowerShellCmdPrefix = "powershell.exe -NonInteractive -ExecutionPolicy Bypass "
+	// command executed on the remote Windows VM. Note that the opening double quote should be closed when using this.
+	remotePowerShellCmdPrefix = "powershell.exe -NonInteractive -ExecutionPolicy Bypass \""
```
Comment (sebsoto) [inline pkg/windows/windows.go:75]: I'd prefer you improve this by adding a new function such as `formatRemotePowerShellCommand(string) string`
which wraps the given string in this prefix and adds the quote at the end.

the trailing quote requirement adds complexity and contributor's/reviewers lives slightly harder

File: docs/wmco-prerequisites.md (hunk 12 above)
Comment (sebsoto) [reply docs/wmco-prerequisites.md:24]: sounds good

File: test/e2e/validation_test.go [hunk 14]
```diff
@@ -299,19 +299,28 @@ func (tc *testContext) sshSetup() error {
...
 			"if ssh -o StrictHostKeyChecking=no -i %s %s@%s 'Get-Help';"+
 				"then CMD_PREFIX=\"\";CMD_SUFFIX=\"\";"+
-				// if PowerShell is not the default shell, explicitly run the command through PowerShell
+				"COMMAND='{"+powershellDefaultCommand+"}';"+
+				// if PowerShell is not the default shell, explicitly run the unmodified command through PowerShell
 				"else CMD_PREFIX=\""+remotePowerShellCmdPrefix+" \\\"\";CMD_SUFFIX=\"\\\"\";"+
+				"COMMAND='{"+command+"}';"+
 				"fi;"+
-				"ssh -o StrictHostKeyChecking=no -i %s %s@%s ${CMD_PREFIX}' %s '${CMD_SUFFIX}",
+				"ssh -o StrictHostKeyChecking=no -i %s %s@%s ${CMD_PREFIX}\" & $COMMAND \"${CMD_SUFFIX}",
```
Comment (sebsoto) [inline test/e2e/validation_test.go:321]: i am mostly to blame but looking at this without this code fresh in my memory, this is very hard to understand

is this a logical AND? what is the function of the &?
this needs to be doc'd better.

File: pkg/windows/windows.go (hunk 13 above)
Comment (mansikulkarni96) [reply pkg/windows/windows.go:75]: +1

File: pkg/windows/windows.go [hunk 15]
```diff
@@ -282,7 +282,7 @@ func (vm *windows) FileExists(path string) (bool, error) {
 
 func (vm *windows) Run(cmd string, psCmd bool) (string, error) {
 	if psCmd && !vm.defaultShellPowerShell {
-		cmd = remotePowerShellCmdPrefix + cmd
+		cmd = remotePowerShellCmdPrefix + cmd + "\""
```
Comment (mansikulkarni96) [inline pkg/windows/windows.go:285]: Are these changes compatible with other supported versions if we backport this?

File: test/e2e/validation_test.go (hunk 14 above)
Comment (mansikulkarni96) [reply test/e2e/validation_test.go:321]: +1 can we have the reasoning behind this added to the commit msg as well?

**Review by mansikulkarni96 — COMMENTED** [review]
Thanks for working on this @saifshaikh48 
PTAL at my comments.

File: pkg/windows/windows.go (hunk 15 above)
Comment (saifshaikh48) [reply pkg/windows/windows.go:285]: Yep, the e2e suite verifies this as we still run 2019 nodes on AWS + Azure.

**Comment by saifshaikh48** [comment]
/retest

Windows to Windows clusterIP networking flake

**Comment by saifshaikh48** [comment]
/hold

Seeing a networking test failure in jobs that use the 2022 image (https://github.com/openshift/windows-machine-config-operator/pull/933 in this PR as well)

**Comment by saifshaikh48** [comment]
Note: we've not copied over the hotfix golden image template to the IBM environment so a vSphere job that lands there will fail.

File: hack/machineset.sh [hunk 16]
```diff
@@ -185,7 +185,7 @@ get_vsphere_ms() {
 
   # set golden image template name
   # TODO: read from parameter
-  template="windows-golden-images/windows-server-2004-template"
+  template="windows-golden-images/windows-server-2022-template-networkHotfix"
```
Comment (jrvaldes) [inline hack/machineset.sh:188]: I don't think the `-networkHotfix` suffix is needed. 

We can stick to `windows-golden-images/windows-server-2022-template` as the latest golden image to avoid code changes. Once the final WS2022 is ready, we just replace the image in vCenter. No PR is needed.

File: test/e2e/network_test.go [hunk 17]
```diff
@@ -443,9 +443,13 @@ func (tc *testContext) getPodIP(selector metav1.LabelSelector) (string, error) {
...
 	var windowsServerImage string
-	if tc.hasCustomVXLAN {
-		// If we're using a custom VXLANPort we will use 2022
+	if tc.CloudProvider.GetType() == config.VSpherePlatformType {
+		// If we're on vSphere we will use 2022
 		windowsServerImage = "mcr.microsoft.com/powershell:lts-nanoserver-ltsc2022"
+	} else if tc.CloudProvider.GetType() == config.NonePlatformType {
+		// TODO: On platform=none we have to use 2004 since the job points to the WS2004 golden image via release repo
+		// When the release repo is updated to use 2022, this clause should be squashed with above - `tc.hasCustomVXLAN`
+		windowsServerImage = "mcr.microsoft.com/powershell:lts-nanoserver-2004"
```
Comment (sebsoto) [inline test/e2e/network_test.go:452]: doesn't seem necessary to do this over 2 different commits
you may have made this a separate one so that this commit can be reverted when the change is made, personally i think doing  	
```if tc.CloudProvider.GetType() == config.VSpherePlatformType && 
tc.CloudProvider.GetType() == config.VSpherePlatformType == config.NonePlatformType {
```
would be better as it makes the platforms used more explicit, instead of obfuscating things with the vxlan port check
so personally i think you should just squash this commit with the previous one

**Review by sebsoto — CHANGES_REQUESTED** [review]

File: hack/machineset.sh (hunk 16 above)
Comment (saifshaikh48) [reply hack/machineset.sh:188]: given `...-networkHotfix` is the name of the VM created in IBM cloud environment, I will stick with this. We will need a PR anyway to move away from pulling 2004 container images for platform=none.

File: test/e2e/network_test.go (hunk 17 above)
Comment (saifshaikh48) [reply test/e2e/network_test.go:452]: @sebsoto I don't understand the suggested code. Shouldn't it be 
```
if tc.CloudProvider.GetType() == config.VSpherePlatformType || 
tc.CloudProvider.GetType() == config.NonePlatformType {
```

**Comment by saifshaikh48** [comment]
/unhold

Golden image replicated to IBM cloud by SPLAT.

File: test/e2e/network_test.go [hunk 18]
```diff
@@ -443,8 +443,12 @@ func (tc *testContext) getPodIP(selector metav1.LabelSelector) (string, error) {
 // getWindowsServerContainerImage gets the appropriate WindowsServer image based on VXLAN port
 func (tc *testContext) getWindowsServerContainerImage() string {
 	var windowsServerImage string
-	if tc.hasCustomVXLAN {
-		// If we're using a custom VXLANPort we need to use 2004
+	if tc.CloudProvider.GetType() == config.VSpherePlatformType {
+		// If we're on vSphere we will use 2022
+		windowsServerImage = "mcr.microsoft.com/powershell:lts-nanoserver-ltsc2022"
+	} else if tc.CloudProvider.GetType() == config.NonePlatformType {
+		// TODO: On platform=none we have to use 2004 since the job points to the WS2004 golden image via release repo
```
Comment (selansen) [inline test/e2e/network_test.go:450]: do we have a ticket to track this one? Don't want this to be left like this without any tracking.

Comment (saifshaikh48) [reply test/e2e/network_test.go:450]: Yes, captured as part of https://issues.redhat.com/browse/WINC-755.
